### PR TITLE
Add the latest metadata design to the design doc.

### PR DESCRIPTION
**Pull request by dcoutts**
Added in an appendix.

### CODE REVIEW

File: shelley/design-spec/delegation_design_spec.tex [hunk 1]
```diff
@@ -4468,6 +4470,162 @@ \subsection{Won't stake pools reject delegation certificates that delegate away
...
+in random access storage. This avoids a problem that Ethereum ran into with
+disk I/O becoming a performance bottleneck. This is why the design does not
+include metadata into the ledger state, and does not make it accessible to
+later scripts.
+
+\subsection{Detail}
+
+The transaction can contain metadata. The metadata hash is part of the body of
+the transaction so is covered by all transaction signatures. The metadata value
+is kept with the transaction witnesses. This follows the `segmented witness'
```
Comment (JaredCorduan) [inline shelley/design-spec/delegation_design_spec.tex]: I'm not sure if "with the transaction witnesses" is potentially confusing. it is indeed outside the body in segwit style, but is separate from the witnesses. hence the "merkle triple".

Comment (dcoutts) [reply shelley/design-spec/delegation_design_spec.tex]: Yeah, I wasn't sure how much detail to go into here. Perhaps I can adjust that to be accurate without being too precise.

Comment (dcoutts) [reply shelley/design-spec/delegation_design_spec.tex]: Pushed an update.

**Pull request merged**